Hardware/Hardware/CardReader/CardReaderSoftwareUpdateState.swift

```diff
@@ -0,0 +1,17 @@
+/// A type that represents the possible states for software update
+public enum CardReaderSoftwareUpdateState {
+    /// An optional update is available
+    case available
+
+    /// A mandatory update has started
+    case started(cancelable: FallibleCancelable?)
+
+    /// The update is being installed
+    case installing(progress: Float)
+
+    /// The update has finished installing
+    case completed
+
+    /// No update available
```

Comment:
What about update failures (i.e. `installationFailed`)? Please open an issue to add that case.

Reply:
Good one: #5036. I missed this initially because the existing code was ignoring errors but was thinking about this as I was working on optional updates and those showed an error message. I'm not sure how we can test the failure case though 🤔

Reply:
> I'm not sure how we can test the failure case though

Could we mock the card reader service update and have it asplode at 0.1 complete?